ql/pricingengines/vanilla/analytichestonengine.hpp

```diff
@@ -99,9 +109,13 @@ namespace QuantLib {
         // Constructor giving full control
         // over the Fourier integration algorithm
         AnalyticHestonEngine(const boost::shared_ptr<HestonModel>& model,
-                             ComplexLogFormula cpxLog, const Integration& itg);
+                             ComplexLogFormula cpxLog, const Integration& itg,
+                             Real andersenPiterbargEpsilon = 1e-8);
 
 
+        // normalized characteristic function
```

Comment:
Maybe this can be in the `private` section?

Reply:
Same goes for chf and lnchf in the other engine.

Reply:
Actually we have tests "against" these functions. IMO we should have the characteristic function of the Heston model and of the piecewise constant Heston model available somewhere. Another place would be the class HestonModel and the class PiecewiseTimeDependentHestonModel but we usuall don't perform complex math in the model classes therefore I've put it into the engine. Now while you bring it up I think we should add a lnChF method to the AnalyticHestonEngine as well for "symmetry" reasons. What do you think?

Reply:
Ok, let's leave them here.